vcf.c

```diff
@@ -1420,6 +1492,7 @@ bcf_hdr_t *bcf_hdr_init(const char *mode)
     if ( (aux->gen = kh_init(hdict))==NULL ) { free(aux); goto fail; }
     aux->key_len = NULL;
     aux->dict = *((vdict_t*)h->dict[0]);
+    aux->version = VCF_DEF;
```

Comment:
This would need to start as something like 0 to allow it to be used to detect if the file version was set.  It should always be set to something else before the value is used because the `fileformat` header is compulsory, but even if it isn't it should still work as we check for version >= `VCF44`.